syncd/PortLinkEventDamper.h

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include <memory>
+
+#include "NotificationHandler.h"
+#include "sairedis.h"
+#include "swss/sal.h"
+#include "swss/select.h"
+#include "swss/selectabletimer.h"
+
+extern "C" {
+#include "saimetadata.h"
+}
+
+namespace syncd
+{
+
+    constexpr uint64_t MICRO_SECS_PER_SEC = 1000000ULL;
+    constexpr uint64_t NANO_SECS_PER_MICRO_SEC = 1000ULL;
+    constexpr uint64_t MICRO_SECS_PER_MILLI_SEC = 1000ULL;
+
+    inline uint64_t getCurrentTimeUsecs()
+    {
+        SWSS_LOG_ENTER();
+
+        struct timespec ts;
+        clock_gettime(CLOCK_MONOTONIC, &ts);
+
+        return (ts.tv_sec * MICRO_SECS_PER_SEC) +
+               (ts.tv_nsec / NANO_SECS_PER_MICRO_SEC);
+    }
+
+  struct DampingStats
+  {
+      // Number of link UP events received.
+      uint64_t pre_damping_up_events;
+      // Number of link DOWN events received.
+      uint64_t pre_damping_down_events;
+      // Number of link UP events advertised post damping.
+      uint64_t post_damping_up_events;
+      // Number of link DOWN events advertised post damping.
+      uint64_t post_damping_down_events;
+      // Timestamp for last advertised link up event post damping.
+      std::string last_advertised_up_event_timestamp;
+      // Timestamp for last advertised link down event post damping.
+      std::string last_advertised_down_event_timestamp;
```

Comment:
add empty line before each line of comment for readibility or put all comments on the right side